Jenkinsfile

```diff
@@ -10,23 +15,54 @@ pipeline {
     }
 
     stages {
+        stage('Pre Build check'){
+            when { branch "PR-*" }
+            steps {
+                script {
+                    pr_number = sh (script: "echo ${env.BRANCH_NAME} | grep -o -E '[0-9]+' ",returnStdout: true)
+                    def json = sh (script: "curl -s https://api.github.com/repos/rook/rook/pulls/${pr_number}", returnStdout: true).trim()
+                    def body = evaluateJson(json,'${json.body}')
+                    if (body.contains("[skip ci]")) {
+                         echo ("'[skip ci]' spotted in PR body text. Aborting.")
+                         env.shouldBuild = "false"
+                    }
+                    if (body.contains("[smoke only]")) {
+                          env.smokeOnly = "true"
+                    }
+                }
+            }
+        }
         stage('Build') {
+            when {
+                expression {
+                    return env.shouldBuild != "false"
```

Comment:
nit: how about `env.shouldBuild == "true"`? it reads easier than the double negative